src/Entity/Unit.php

```diff
@@ -382,4 +385,33 @@ public static function baseFieldDefinitions(EntityTypeInterface $entity_type) :
     return $fields;
   }
 
+  /**
+   * Get comma separated list of provided language names.
+   *
+   * @return string
+   *   Comma separated list of language names.
+   */
+  public function getProvidedLanguageNames(): string {
+    $provided_languages = $this->get('provided_languages');
+    if ($provided_languages->isEmpty()) {
+      return '';
+    }
+
+    $all_languages = $this->languageManager()->getStandardLanguageList();
+
+    $language_names = [];
+    foreach ($provided_languages as $provided_language) {
```

Comment:
I think this can be simplified with something like:

```php
if (!$language = $this->languageManager()->getLanguage($langcode)) {
  continue;
}
```

Reply:
Nope, getLanguage seems to only return a language object if the given language is enabled.